update_image.pl

```diff
@@ -112,10 +144,206 @@
+}
+else {
+    run_command("cp $hb_binary_dir/$sbe_binary_filename $scratch_dir/");
+}
+
+sub processConvergedSections {
+
+    use constant EMPTY => "EMPTY";
+
+    my $stop_basename = $wink_binary_filename;
+    $stop_basename =~ s/.hdr.bin.ecc//;
+
+    my $sbePreEcc = "$scratch_dir/$sbe_binary_filename";
+    $sbePreEcc =~ s/.ecc//;
+
+    # Source and destination file for each supported section
+    my %sections=();
+    $sections{HBBL}{in}        = "$scratch_dir/hbbl.bin";
+    $sections{HBBL}{out}       = "$scratch_dir/hbbl.bin.ecc";
+    $sections{HBB}{in}         = "$hb_image_dir/img/hostboot.bin";
+    $sections{HBB}{out}        = "$scratch_dir/hostboot.header.bin.ecc";
+    $sections{HBI}{in}         = "$hb_image_dir/img/hostboot_extended.bin";
+    $sections{HBI}{out}        = "$scratch_dir/hostboot_extended.header.bin.ecc";
+    $sections{HBD}{in}         = "$op_target_dir/$targeting_binary_source";
+    $sections{HBD}{out}        = "$scratch_dir/$targeting_binary_filename";
+    $sections{SBE}{in}         = "$sbePreEcc";
+    $sections{SBE}{out}        = "$scratch_dir/$sbe_binary_filename";
+    $sections{PAYLOAD}{in}     = "$payload.bin";
+    $sections{PAYLOAD}{out}    = "$scratch_dir/$payload_filename";
+    $sections{SBKT}{out}       = "$scratch_dir/SBKT.bin";
+    $sections{HCODE}{in}       = "$hb_binary_dir/${stop_basename}.bin";
+    $sections{HCODE}{out}      = "$scratch_dir/${stop_basename}.hdr.bin.ecc";
+    $sections{HBRT}{in}        = "$hb_image_dir/img/hostboot_runtime.bin";
+    $sections{HBRT}{out}       = "$scratch_dir/hostboot_runtime.header.bin.ecc";
+    $sections{OCC}{in}         = "$occ_binary_filename";
+    $sections{OCC}{out}        = "$occ_binary_filename.ecc";
+    $sections{BOOTKERNEL}{in}  = "$binary_dir/$bootkernel_filename";
+    $sections{BOOTKERNEL}{out} = "$scratch_dir/$bootkernel_filename";
+    $sections{CAPP}{in}        = "$capp_binary_filename";
+    $sections{CAPP}{out}       = "$scratch_dir/cappucode.bin.ecc";
+
+    if(-e $wof_binary_filename)
+    {
+        $sections{WOFDATA}{in} = "$wof_binary_filename";
+    }
+    else
+    {
+        print "ERROR: WOFDATA partition is not found, including blank binary instead\n";
+    }
+    $sections{WOFDATA}{out}     = "$scratch_dir/wofdata.bin.ecc";
+
+    # Build up the system bin files specification
+    my $system_bin_files;
+    foreach my $section (keys %sections)
+    {
+        if(exists $sections{$section}{in})
+        {
+            $_ = $sections{$section}{in};
+            if((/ecc/i) || (/pad/i))
+            {
+                die "Input file's name, $sections{$section}{in}, suggests padding "
+                    . "or ECC, neither of which is allowed.";
+            }
+        }
+
+        my $separator = length($system_bin_files) ? "," : "";
```

Comment:
I added this comment above it .. the only purpose is to prevent adding a leading comma separator when it's the first element of the bin file specification, it has nothing to do with system/chip names.
        # If the system bin files specification has nothing in it yet, avoid
        # adding a separator